crates/ide/src/inlay_hints.rs

```diff
@@ -299,6 +305,26 @@ fn is_argument_similar_to_param_name(
     }
 }
 
+fn is_param_name_similar_to_fn_name(
+    param_name: &str,
+    param_num: usize,
+    fn_name: Option<&String>,
+) -> bool {
+    // if it's the first parameter, don't show it if:
+    // - is the same as the function name, or
+    // - the function ends with '_' + param_name
+
+    match (param_num, fn_name) {
+        (0, Some(function)) => {
+            function == param_name
+                || (function.len() > param_name.len()
+                    && function[..function.len() - param_name.len()].ends_with('_')
+                    && function.ends_with(param_name))
```

Comment:
```suggestion
                    && function.ends_with(param_name))
                    && function[..function.len() - param_name.len()].ends_with('_')
```

otherwise, this'll panic for wide characters with invalid utf-8 boundary.